stdlib/src/builtin/file_descriptor.mojo

```
@@ -85,6 +88,35 @@ struct FileDescriptor(Writer):
                 written,
             )
 
+    @always_inline
+    fn read_bytes(mut self, size: UInt) raises -> List[Byte]:
```

Are we sure we don't want this to be:
`fn read_bytes(mut self, mut buffer: Span[Byte, _]) raises -> UInt`? 

I'm not sure we want to force allocations for basic IO, and this allows partial reads. You can just resize the list.

---

Yeah +1, when doing several reads in a loop this will avoid having many allocs for some custom buffered stuff. Also when you just want to write to some other destination directly

---

+1 to not allocating for users on the return type.  Having this just return a `UInt` allows for partial reads which is important too.  This is how we plan on modeling a `Readable` trait in the future too.

When POSIX `read` fails (when it returns `-1`), we can just raise an error based on the errno value.

---

Really good point on not allocating , to avoid n+1 allocing!